### ISSUE TITLE
BUG: Out of range float values are not JSON compliant

### ISSUE THREAD

**Issue by mrembalski** [neptune-ai/neptune-client#1351]
neptune-client==1.2.0,
python 3.10.10

It seems that Neptune does not properly handle things that are inside a batch (I'm using pretokenized inputs, so my batch consists of tensors).

```
Exception in thread NeptuneAsyncOpProcessor:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/operation_processors/async_operation_processor.py", line 230, in run
    super().run()
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/threading/daemon.py", line 53, in run
    self.work()
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/operation_processors/async_operation_processor.py", line 246, in work
    self.process_batch([element.obj for element in batch], batch[-1].ver)
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/threading/daemon.py", line 76, in wrapper
    result = func(self_, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/operation_processors/async_operation_processor.py", line 259, in process_batch
    processed_count, errors = self._processor._backend.execute_operations(
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/backends/hosted_neptune_backend.py", line 484, in execute_operations
    self._execute_operations(
  File "/usr/local/lib/python3.10/site-packages/neptune/common/backends/utils.py", line 65, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/backends/hosted_neptune_backend.py", line 642, in _execute_operations
    result = self.leaderboard_client.api.executeOperations(**kwargs).response().result
  File "/usr/local/lib/python3.10/site-packages/neptune/internal/backends/swagger_client_wrapper.py", line 96, in __call__
    future = self._api_method(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/bravado/client.py", line 271, in __call__
    request_params = construct_request(
  File "/usr/local/lib/python3.10/site-packages/bravado/client.py", line 312, in construct_request
    construct_params(operation, request, op_kwargs)
  File "/usr/local/lib/python3.10/site-packages/bravado/client.py", line 335, in construct_params
    marshal_param(param, param_value, request)
  File "/usr/local/lib/python3.10/site-packages/bravado_core/param.py", line 155, in marshal_param
    request['data'] = json.dumps(value)
  File "/usr/local/lib/python3.10/site-packages/simplejson/__init__.py", line 378, in dumps
    return _default_encoder.encode(obj)
  File "/usr/local/lib/python3.10/site-packages/simplejson/encoder.py", line 298, in encode
    chunks = self.iterencode(o)
  File "/usr/local/lib/python3.10/site-packages/simplejson/encoder.py", line 379, in iterencode
    return _iterencode(o, 0)
ValueError: Out of range float values are not JSON compliant
```

**Comment by MirenLurBarquin** [comment]
I have been facing the same issue, could it be that your tensor contains non JSON compliant (e.g., NaN or Inf) values?

**Comment by Blaizzy** [comment]
Hi @mrembalski 

Prince here,

Thanks for reaching out!

Could you share a minimal reproducible example?

**Comment by Blaizzy** [comment]
In that case, you are correct @MirenLurBarquin!

We don't support NaN and Inf, we recommend replacing them with Zero.

**Comment by SiddhantSadangi** [comment]
Hello @mrembalski ,

Can you share your version of `simplejson`? This issue seems to be caused by `simplejson>=3.19`. 
Downgrading to 3.18 should solve the issue.

Neptune will still not log `nan`/`inf` values. Downgrading simplejson will show a warning from the server if such values are encountered instead of crashing the entire thread.

Please let me know if it works for you.

**Comment by mrembalski** [comment]
@SiddhantSadangi yeah, downgrading works. 

The error message could be kind of confusing - I thought that neptune was somehow handling/logging batch data instead of metrics (in my case). 

It could be just shown, for what key (since it was in a JSON dict) it failed. That would be very helpful. 

TL;DR if someone stumbles upon it in the future - PRAUC from `torchmetrics` returns NaN, when there are no positives in a batch. Then Neptune Logger (through `simplejson`) fails. Either use `torch.nan_to_num` or don't log it.

**Comment by SiddhantSadangi** [comment]
Thanks for the TL;DR and the workaround @mrembalski ! 🙏 

I've also passed on the feedback about the error message to the developers.

I'll be closing this issue now. Please feel free to contact us through any of our support channels if you have any suggestions or queries.
We are always happy to hear and help 🤗
